edgraph/server.go

```diff
@@ -1894,6 +1894,218 @@ func (s *Server) CheckVersion(ctx context.Context, c *api.Check) (v *api.Version
 	return v, nil
 }
 
+func (s *Server) CreateNamespace(ctx context.Context, in *api.CreateNamespaceRequest) (
+	*api.CreateNamespaceResponse, error) {
+
+	if err := AuthGuardianOfTheGalaxy(ctx); err != nil {
+		s := status.Convert(err)
+		return nil, status.Error(s.Code(),
+			"Non guardian of galaxy user cannot create namespace. "+s.Message())
+	}
+
+	if err := verifyNamespaceName(in.NsName); err != nil {
+		return nil, err
+	}
+
+	if _, err := getNamespaceID(ctx, in.NsName); err == nil {
+		return nil, errors.Errorf("namespace %q already exists", in.NsName)
+	} else if !strings.Contains(err.Error(), "not found") {
+		return nil, err
+	}
+
+	password := "password"
```

Comment:
Why do we need a default password here?

Reply:
In case someone passes ab empty string here.